pkgs/development/libraries/ffmpeg/generic.nix

```
@@ -472,6 +472,11 @@ stdenv.mkDerivation (finalAttrs: {
         url = "https://gitlab.archlinux.org/archlinux/packaging/packages/ffmpeg/-/raw/a02c1a15706ea832c0d52a4d66be8fb29499801a/add-av_stream_get_first_dts-for-chromium.patch";
         hash = "sha256-DbH6ieJwDwTjKOdQ04xvRcSLeeLP2Z2qEmqeo8HsPr4=";
       })
+      # Fix https://github.com/NixOS/nixpkgs/issues/383210
+      (fetchpatch2 {
+        url = "https://github.com/FFmpeg/FFmpeg/commit/099f88b8641dfc299f3896d17d9addc5b9ae7799.patch";
+        hash = "sha256-T9wmV1r0IOnV35pii6JkMkHMs3d4v1Ap1/Hd8FPi/pE=";
```

```suggestion
      (fetchpatch {
        url = "https://github.com/FFmpeg/FFmpeg/commit/099f88b8641dfc299f3896d17d9addc5b9ae7799.patch";
        hash = "sha256-xxx=";
```

don't use fetchpatch2 unless it is required which it isn't in this case. it is unstable without appending `?full_index=1` to the url given that the line with `index` can change over time.

---

the hash needs to be updated as well. `sha256-OzRWpkxmY9UlnBR0w3hL4wykVVc0HJfw5Sefh3uJAsg=` or something -- `fetchpatch2` and `fetch patch` hashes aren't compatible. I didn't realize that all of the other patches are using `fetchpatch2` but they all have the same problem. hopefully the repository isn't too active.

---

We should probably use `https://git.ffmpeg.org/gitweb/ffmpeg.git/patch/` rather than the GitHub mirror, as with the other patches. (I don’t know if Gitweb has the same “lengthening hashes” problem as GitHub does. Probably it does.)

---

gitweb it is also truncating the hashes so assume it has the same issue as GitHub.

---

Thanks for the quick replies! I've changed the patch url from `github.com` to `git.ffmpeg.org`.

(To keep a clean diff, maybe it's best not to modify the existing `fetchpatch2` instances. If `fetchpatch2` is considered deprecated, then maybe the instances of `fetchpatch2` across nixpkgs can be refactored in a separate PR. This would require replacing `fetchpatch2` by `fetchpatch` and updating all the hashes.)

---

The Gitweb patches still have the `index …` lines that `fetchpatch` strips but `fetchpatch2` doesn’t, and that are likely nondeterministic over time. We should probably switch to `fetchpatch` for patches that don’t involve renames, but it’s orthogonal to this PR.